### PR TITLE
Fix #381: Add GeoTrust to Pushy client

### LINKED ISSUE #381: Add GeoTrust to Pushy client
See the following issues for details:

- https://github.com/jchambers/pushy/issues/809
- https://github.com/jchambers/pushy/issues/833

The new Java version does not contain a GeoTrust certificate and hence calls fail on SSL handshake.

### PR DESCRIPTION
**Pull request by petrdvorak**

### CODE REVIEW

File: powerauth-push-server/src/main/java/io/getlime/push/util/CaCertUtil.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+            result.add(geoTrustCert);
+        } catch (CertificateException | IOException e) {
+            logger.error("Certificate error: {}", e.getMessage(), e);
+        }
+
+        return result.toArray(new X509Certificate[0]);
+    }
+
+    private static X509Certificate certificateFromPem(String pem) throws CertificateException {
+        byte [] decoded = BaseEncoding.base64().decode(pem
```
Comment (romanstrobl) [inline powerauth-push-server/src/main/java/io/getlime/push/util/CaCertUtil.java:83]: Strange formatting - should be `byte[]`.

File: powerauth-push-server/src/main/java/io/getlime/push/util/CaCertUtil.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        } catch (CertificateException | KeyStoreException | NoSuchAlgorithmException | InvalidAlgorithmParameterException | IOException e) {
+            logger.error("Certificate error: {}", e.getMessage(), e);
+        }
+
+        // Attempt to add at least GeoTrust Global CA required by Apple for APNs
+        try {
+            final File resource = new ClassPathResource("cacert/GeoTrust_Global_CA.pem").getFile();
+            final String geoTrustCertString = new String(Files.readAllBytes(resource.toPath()));
+            final X509Certificate geoTrustCert = certificateFromPem(geoTrustCertString);
+            result.add(geoTrustCert);
```
Comment (romanstrobl) [inline powerauth-push-server/src/main/java/io/getlime/push/util/CaCertUtil.java:74]: What happens in Java 8, won't the certificate be present twice?

**Review by romanstrobl — APPROVED** [review]
Thanks for fixing the certificate issue. I have 2 minor comments.

File: powerauth-push-server/src/main/java/io/getlime/push/util/CaCertUtil.java (hunk 1 above)
Comment (petrdvorak) [reply powerauth-push-server/src/main/java/io/getlime/push/util/CaCertUtil.java]: This is fixed.

File: powerauth-push-server/src/main/java/io/getlime/push/util/CaCertUtil.java (hunk 2 above)
Comment (petrdvorak) [reply powerauth-push-server/src/main/java/io/getlime/push/util/CaCertUtil.java:74]: I think this is handled at some point by the `equals` call - in any case, it works on:

```
$ java -version
java version "1.8.0_231"
Java(TM) SE Runtime Environment (build 1.8.0_231-b11)
Java HotSpot(TM) 64-Bit Server VM (build 25.231-b11, mixed mode)
```

**Pull request merged**